components/applications-service/pkg/storage/client.go

```diff
@@ -42,6 +42,36 @@ const (
 	None     = "NONE"
 )
 
+type UpdateStrategy int
+
+const (
+	NoneStrategy UpdateStrategy = iota
```

Comment:
💭 could this cause an issue if, say habitat adds a new kind of upgrade strategy and the user adopts the new upgrade strategy before they upgrade automate to a newer version that has this "enum" updated? If we really want this to be an enum then maybe `0` should be "Unknown" or something like that(?). "NONE" implies that the supervisor won't update following a package promotion, which could affect a use case like "find all the services that will update if I promote package X to the "production" channel.

Reply:
No. When habitat has a new upgrade strategy, the habitat team has to also update the protobuf enum defined in the event messages, that is where automate and habitat won’t be in sync and we need to update our protobuf and code to accept the new strategy. 

Going deeper into this topic, any modification that habitat will do will also trigger a need for us to update our protobuf and code. Unless we code to accept anything that comes through, which has its trade offs. 

What would you like do to here? Remove the enum or continue having it?